### PR TITLE
Token Review

### PR DESCRIPTION
**Pull request by shawn-hurley**
Adding ability to us a middleware to validate the bearer token from the service catalog. 

* uses https://github.com/shawn-hurley/osb-broker-k8s-lib 
* brings in dependencies on client-go, apimachinery and their dependencies.

### CODE REVIEW

**Comment by shawn-hurley** [comment]
One of the issues that I had trouble solving was how to test the token review. I would like to use the client-go mock to check but don't know how to set it for the token review middleware. Any ideas?

**Comment by LiliC** [comment]
In case you haven't seen it, there is also this project that does something similar https://github.com/brancz/kube-rbac-proxy

File: cmd/servicebroker/main.go [hunk 1]
```diff
@@ -22,15 +26,19 @@ import (
...
+	TLSKey       string
+	Authenticate bool
+	KubeConfig   string
 }
 
 func init() {
 	flag.IntVar(&options.Port, "port", 8005, "use '--port' option to specify the port for broker to listen on")
 	flag.StringVar(&options.TLSCert, "tlsCert", "", "base-64 encoded PEM block to use as the certificate for TLS. If '--tlsCert' is used, then '--tlsKey' must also be used. If '--tlsCert' is not used, then TLS will not be used.")
 	flag.StringVar(&options.TLSKey, "tlsKey", "", "base-64 encoded PEM block to use as the private key matching the TLS certificate. If '--tlsKey' is used, then '--tlsCert' must also be used")
+	flag.BoolVar(&options.Authenticate, "authenticate", false, "option to specify if the broker should validate the bearer auth token with kubernetes")
```
Comment (pmorie) [inline cmd/servicebroker/main.go]: We should call this CLI flag `authenticate-k8s-token` or something. CF only currently supports basic auth.

**Comment by shawn-hurley** [comment]
@pmorie I think this should be ready for another review

**Comment by pmorie** [comment]
LGTM - I'm going to try to this out locally before I merge.

**Pull request merged**